content/source/docs/extend/terraform-0.12-compatibility.html.md

```diff
@@ -0,0 +1,329 @@
+---
+layout: "extend"
+page_title: "Terraform 0.12 Compatibility for Providers"
+sidebar_current: "docs-extend-0.12-compatibility"
+description: |-
+  Compatibility with Terraform 0.12 requires some changes to existing provider
+  codebases.
+---
+
+# Terraform 0.12 Compatibility for Providers
+
+Terraform 0.12 introduced a new type system for the Terraform language, and
+with it some changes to the representations of configuration, state, and plans.
+To support these changes, Terraform 0.12 introduced a new protocol for Terraform
+Core to interact with providers.
+
+The _provider protocol_ is the physical mechanism by which Terraform Core
+launches a provider executable and directs it to take actions. Provider
+developers do not generally interact with the protocol directly, but rather
+implement against the Terraform SDK (as described elsewhere in
+[the _Extend_ section](./)) which in turn implements the provider side of
+the protocol itself.
+
+As a result, most of the work to support Terraform 0.12 comes just from
+upgrading to the latest Terraform SDK, which features support for both the
+old and new provider protocols.
+
+Although the SDK aims to abstract over as many of the differences as possible,
+the changes to the Terraform language were significant and so in practice
+some small adjustments to provider code may be required to ensure the best
+possible compatibility with Terraform 0.12. The goal of this guide is to
+describe some common situations and how to address them.
+
+-> Terraform internally uses a separate versioning scheme for the provider
```

Comment:
Callout should include a note/word: https://github.com/hashicorp/terraform-website/blob/master/README.md#callouts ... I might say this is more an Important than a Note

Reply:
I chose "note" here because you don't really need to know this information unless you happen to see the error message that talks about it. It doesn't really feel "important" or "warning" to me, but I don't feel too strongly about it. :thinking: